### PR TITLE
Conditionally add the graceful shutdown Kubelet parameters

### PR DESCRIPTION
**Pull request by torredil**
#### What this PR does / why we need it:

This PR introduces changes to conditionally add the graceful shutdown Kubelet parameters `shutdownGracePeriod` and `shutdownGracePeriodCriticalPods`, which can only be set via the [configuration file](https://kubernetes.io/blog/2021/04/21/graceful-node-shutdown-beta/#how-do-i-use-it) as CLI flags aren't added for new Kubelet features: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/options.go

This change helps facilitate the testing of the graceful shutdown feature in full cluster e2e tests.

#### Which issue(s) this PR fixes:

Fixes # N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -778,6 +778,14 @@ function construct-linux-kubelet-flags {
     flags+="  --image-credential-provider-bin-dir=${AUTH_PROVIDER_GCP_LINUX_BIN_DIR}"
   fi
 
+  # Conditionally add the graceful shutdown flags
```
Comment (aojea) [inline cluster/gce/util.sh:781]: if you see line 805

>   else # For nodes
>    flags+=" ${NODE_KUBELET_TEST_ARGS:-}"

you can already pass custom flags to the kubelet in the jobs

Comment (torredil) [reply cluster/gce/util.sh:781]: Okay, I've updated https://github.com/kubernetes/test-infra/pull/32728 to pass the custom flags using`NODE_KUBELET_TEST_ARGS`.

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -1029,6 +1029,18 @@ authentication:
   x509:
     clientCAFile: /etc/srv/kubernetes/pki/ca-certificates.crt
 EOF
+
```
Comment (aojea) [inline cluster/gce/util.sh:1032]: This looks correct, we talked offline and it seems the only way to configure these options is via config file

@BenTheElder is this the right way?

**Pull request merged**